lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -344,10 +343,11 @@ function _campaign_resource_reportback($nid, $values) {
     }
   }
 
+  $values['fid'] = $file->fid;
+  $values['file'] = dosomething_helpers_get_data_uri_from_fid($file->fid);
+
   // @todo: Move this logic into dosomething_reportback_save().
-  if ($file) {
```

Comment:
You need this conditional before you do anything with `$file` to make sure that a file was added